cmd/gencerts_test.go

```diff
@@ -0,0 +1,122 @@
+package cmd_test
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/OpenBazaar/openbazaar-go/cmd"
+	"github.com/OpenBazaar/openbazaar-go/schema"
+)
+
+func buildCertDirectory() (*string, func(), error) {
+	ctx := schema.SchemaContext{
+		DataPath:        schema.GenerateTempPath(),
+		TestModeEnabled: true,
+	}
+	certSchema := schema.MustNewCustomSchemaManager(ctx)
+	if err := certSchema.BuildSchemaDirectories(); err != nil {
+		return nil, nil, err
+	}
+	return &ctx.DataPath, certSchema.DestroySchemaDirectories, nil
+}
+
+func TestGenCertsGenericDefaults(t *testing.T) {
+	var dataPath, Destroy, schema_err = buildCertDirectory()
+	dataDir := string([]byte(*dataPath))
+	if schema_err != nil {
+		Destroy()
+		t.Fatal(schema_err)
+	}
+
+	config := cmd.GenerateCertificates{
+		DataDir:  dataDir,
+		Testnet:  true,
+		Host:     "127.0.0.1",
+		ValidFor: 1, // 1ns ... 1*1e9 == 1s
+	}
+
+	args := []string{"args?"}
```

Comment:
FYI, Execute satisfies the Commander interface defined here: https://github.com/jessevdk/go-flags/blob/c0bdd90e3f6f784046c4b384a3e78770aaa31b07/command.go#L41

It would have the specific args passed in from the command line.